taskcluster/homebrew-build.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+set -xe
+
+OS=$(uname)
+if [ "${OS}" != "Darwin" ]; then
+    echo "This should only run on OSX."
+    exit 1
+fi;
+
+flavor=$1
+
+source $(dirname "$0")/tc-tests-utils.sh
+
+if [ -z "${TASKCLUSTER_TASK_DIR}" ]; then
+  echo "No TASKCLUSTER_TASK_DIR, aborting."
+  exit 1
+fi
+
+do_prepare_homebrew()
+{
+  local _brew_instance=$1
+
+  if [ -z "${_brew_instance}" ]; then
+    echo "No _brew_instance, aborting."
+    exit 1
+  fi
+
+  export PATH=${_brew_instance}/bin:$PATH
+  export HOMEBREW_LOGS="${_brew_instance}/homebrew.logs/"
+  export HOMEBREW_CACHE="${_brew_instance}/homebrew.cache/"
+
+  # Never fail on pre-existing homebrew/ directory
+  mkdir -p "${_brew_instance}" || true
+  mkdir -p "${HOMEBREW_CACHE}" || true
+
+  # Make sure to verify there is a 'brew' binary there, otherwise install things.
+  if [ ! -x "${_brew_instance}/bin/brew" ]; then
+    curl -L ${BREW_URL} | tar xz --strip 1 -C "${_brew_instance}"
```

Comment:
Maybe we should use this opportunity to also pin the formulas? That way we avoid breakage when the formula language changes in a backwards incompatible way.

Reply:
For example by saving a `${BREW_FORMULAS_COMMIT}` and then checking out that commit in `` `brew --prefix`/Homebrew/Library/Taps/homebrew/homebrew-core``

Reply:
> Maybe we should use this opportunity to also pin the formulas? That way we avoid breakage when the formula language changes in a backwards incompatible way.

Last that I checked, this was explicitely not supported. Aren't we safe with version-URL + the no auto update env var ?

Reply:
The problem is that when you install it and run it, it clones the master branch of the formulas repo, regardless of the version of Homebrew itself.

Reply:
> The problem is that when you install it and run it, it clones the master branch of the formulas repo, regardless of the version of Homebrew itself.

Honestly, homebrew is a mess.